File: api/payloads/address.go
package payloads

type Address struct {
	ID          uint    `json:"id"`
	Name        string  `json:"name"`
	Region      Region  `json:"region" binding:"required"`
	City        string  `json:"city" binding:"required"`
	Zip         string  `json:"zip" binding:"required"`
	Address1    string  `json:"address1" binding:"required"`
	Address2    *string `json:"address2"`
	PhoneNumber string  `json:"phoneNumber"`
}

File: src/core/types/address.js

type State = {
  name: string,
};

type Country = {
  alpha3: string,
  name?: string,
};

type Region = {
  id: number,
  countryId: number,
  name: string,
};

export type Address = {
  id?: number,
  region?: Region,
  country?: Country,
  state?: State,
  city?: string,
  name: string,
  address1: string,
  address2?: string,
  zip: string|number,
  isDefault?: boolean,
  phoneNumber?: string,
  isDeleted?: boolean,
};
